source/uwp/Renderer/lib/Util.h

```diff
@@ -234,98 +193,54 @@ struct DECLSPEC_UUID("defc7d5f-b4e5-4a74-80be-d87bd50a2f45") ITypePeek : IInspec
     virtual ~ITypePeek() {}
 };
 
-template<typename T, typename R> Microsoft::WRL::ComPtr<T> PeekInnards(R r)
+template<typename D, typename I> winrt::com_ptr<D> peek_innards(I&& o)
 {
-    Microsoft::WRL::ComPtr<T> inner;
-    Microsoft::WRL::ComPtr<ITypePeek> peeker;
-
-    if (r && SUCCEEDED(r->QueryInterface(__uuidof(ITypePeek), &peeker)))
+    winrt::com_ptr<D> out;
+    if (auto p = o.try_as<ITypePeek>())
     {
-        inner = reinterpret_cast<T*>(peeker->PeekAt(__uuidof(T)));
+        if (p->PeekAt(__uuidof(D)))
+        {
+            out.copy_from(winrt::get_self<D>(o));
+        }
     }
-    return inner;
+
+    return out;
 }
-void GetUrlFromString(_In_ ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveHostConfig* hostConfig,
-                      _In_ HSTRING urlString,
-                      _Outptr_ ABI::Windows::Foundation::IUriRuntimeClass** url);
 
-ABI::Windows::UI::Color GenerateLHoverColor(const ABI::Windows::UI::Color& originalColor);
+winrt::Uri GetUrlFromString(winrt::AdaptiveHostConfig const& hostConfig,
+                                                 winrt::hstring const& urlString);
 
-ABI::Windows::Foundation::DateTime GetDateTime(unsigned int year, unsigned int month, unsigned int day);
+winrt::Windows::UI::Color GenerateLHoverColor(winrt::Windows::UI::Color const& originalColor);
 
-HRESULT GetDateTimeReference(unsigned int year,
-                             unsigned int month,
-                             unsigned int day,
-                             _COM_Outptr_ ABI::Windows::Foundation::IReference<ABI::Windows::Foundation::DateTime>** dateTimeReference);
+winrt::DateTime GetDateTime(unsigned int year, unsigned int month, unsigned int day);
 
-HRESULT CopyTextElement(_In_ ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveTextElement* textElement,
-                        _COM_Outptr_ ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveTextElement** copiedTextElement);
+winrt::IReference<winrt::DateTime> GetDateTimeReference(unsigned int year,
+                                                                                                  unsigned int month,
+                                                                                                  unsigned int day);
 
-namespace AdaptiveCards::Rendering::Uwp
+winrt::IAdaptiveTextElement CopyTextElement(winrt::IAdaptiveTextElement const& textElement);
+
+// TODO: rethink this perhaps?
```

Comment:
@paulcam206 could you please have a look at this?

Reply:
is this even referenced anywhere?

Reply:
https://github.com/microsoft/AdaptiveCards/blob/user/vsiliush/renderer-winrt-final/source/uwp/Renderer/lib/TextHelpers.cpp#L34-L47

Reply:
was having some search issues, obviously... I think this is fine for now...